zanata-war/etc/scripts/Mysql2Postgresql.groovy

```diff
@@ -0,0 +1,457 @@
+    columns.close()
+}
+mysqlTables.close()
+
+// Create indexes
+mysqlTables = mysqlMetadata.getTables(null, null, null, ['TABLE'] as String[])
+while(mysqlTables.next()) {
+    def tableName = mysqlTables.getString(3)
+    ResultSet indexes = mysqlMetadata.getIndexInfo(null, null, tableName, false, false)
+
+    createIndexes(tableName, indexes)
+
+    indexes.close()
+}
+mysqlTables.close()
+
+// Create data (only if migrating directly to a DB)
+if(!generateDDL) {
+    mysqlTables =
+        mysqlMetadata.getTables(null, null, null, ['TABLE'] as String[])
+    while (mysqlTables.next()) {
+        def tableName = mysqlTables.getString(3)
+        try {
+            extractDataForTable(tableName)
+        }
+        catch (SQLException ex) {
+            if (ex.nextException) {
+                throw ex.nextException
+            } else {
+                throw ex
+            }
+        }
+    }
+    mysqlTables.close()
+}
+
+// Create All Foreign Keys
+mysqlTables = mysqlMetadata.getTables(null, null, null, ['TABLE'] as String[])
+while(mysqlTables.next()) {
+    def tableName = mysqlTables.getString(3)
+    ResultSet foreignKeys = mysqlMetadata.getImportedKeys(null, null, tableName)
+
+    createForeignKeys(tableName, foreignKeys)
+
+    foreignKeys.close()
+}
+mysqlTables.close()
+
+// Create Triggers
+createTriggers()
+
+// Manually increment sequences (only if migrating directly to a DB)
+if(!generateDDL) {
+    updateAutoIncrementSequences(autoIncrementSeqs)
+}
+
+// ========================= Helper Functions =========================
+
+// Initialize all necessary variables, using parameters of default values where
+// indicated
+// Returns an object indicating if the script should proceed
+def initialize(args) {
+    def cli = new CliBuilder(usage: 'groovy Mysql2Postgresql.groovy [options]', header: 'Options:')
+
+    cli.mysqlhost(args:1, argName: 'mysqlhost', 'Host name or IP address of the mysql database (default: localhost)')
+    cli.mysqlport(args:1, argName: 'mysqlport', 'Open port to connect to the mysql database (default: 3306)')
+    cli.mysqlschema(args:1, argName: 'mysqlschema', 'MySql schema to migrate')
+    cli.mysqluser(args:1, argName: 'mysqluser', 'User for the mysql database (default: "")')
+    cli.mysqlpassword(args:1, argName: 'mysqlpassword', 'Password for the mysql database (default: "")')
+
+    cli.psqlhost(args:1, argName: 'psqlhost', 'Host name or IP address of the postgresql database (default: localhost)')
+    cli.psqlport(args:1, argName: 'psqlport', 'Open port to connect to the postgresql database (default: 5432)')
+    cli.psqldb(args:1, argName: 'psqldb', 'Postgresql database to migrate data into')
+    cli.psqluser(args:1, argName: 'psqluser', 'User for the portgresql database (default: "")')
+    cli.psqlpassword(args:1, argName: 'psqlpassword', 'Password for the postgresql database (default: "")')
+
+    cli.script(args:1, argName: 'script', 'Script file with the PostgreSQL DDL')
```

Comment:
Should say something like "Output a script file with...". It's a bit unclear as to what it does.